src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -135,7 +159,12 @@ public override FormulaValue Visit(CallNode node, SymbolContext context)
 
             var childContext = context.WithScope(node.Scope);
 
-            if (func is CustomTexlFunction customFunc)
+            if (func is IAsyncTexlFunction asyncFunc)
+            {
+                var result = await asyncFunc.InvokeAsync(args, _cancel);
+                return result;
+            }
```

Comment:
This is the critical change - ability to invoke an async function. 
everything else is viral from this.